### DISCUSSION TITLE
Possibilitty to return element from pop or shift function in use-list-state hook

### DISCUSSION THREAD

**Discussion by bastiengrignon** [mantinedev/mantine#4228] [Feature requests]
I think it could be nice to have the **poped** or **shifted** element returned instead of `void`. 
Or it does not respect some kind of guidline or good practice :thinking:

**Comment by rtivital** [comment]
pop and shift methods use callback function to modify state, it is impossible to return removed item – https://github.com/mantinedev/mantine/blob/master/src/mantine-hooks/src/use-list-state/use-list-state.ts#L39:L51

**Comment by bastiengrignon** [comment]
Ok too bad but thanks for your response and all your work :wink:
